### ISSUE TITLE
Error on Mac

### ISSUE THREAD

**Issue by sonyamann** [ZcashFoundation/zec-qt-wallet#58]
> zcashd said: Error: Cannot find the Zcash network parameters in the following directory:
"/Users/sonya/Library/Application Support/ZcashParams"
> Please run 'zcash-fetch-params' or './zcutil/fetch-params.sh' and then restart.

Just downloaded and opened zec-qt-wallet on my Mac for the first time. Not sure why I got this error — apparently related to zcashd, but I was under the impression that the zec-qt-wallet would handle that for me? I'm on macOS High Sierra 10.13.6, if that's helpful.

**Comment by garethtdavies** [comment]
I'm curious have you ever run zcashd on this Mac before? Do you have an existing conf file in `"/Users/sonya/Library/Application Support/Zcash/zcash.conf"`? 

That error is from zcashd which is shown by zec-qt-wallet but yeh it should (and does normally) download the params.

**Comment by adityapk00** [comment]
I'm guessing that zec-qt-wallet didn't download the params (it should have), which is why you're seeing this error. 

Can you check a couple of things for me?
- Is there anything in the `/Users/sonya/Library/Application Support/ZcashParams` directory? 
- Do you have a zcash.conf in `/Users/sonya/Library/Application Support/Zcash/zcash.conf`?

**Comment by sonyamann** [comment]
In `/Users/sonya/Library/Application Support/ZcashParams`:

* `sapling-output.params`
* `sapling-spend.params`
* `sprout-groth16.params`
* `sprout-verifying.key`

And yep, `/Users/sonya/Library/Application Support/Zcash/` contains `zcash.conf`.

**Comment by adityapk00** [comment]
Very interesting. For some reason, you're missing the `sprout-proving.key` from `https://z.cash/downloads/sprout-proving.key`

I wonder why. I'm working on a build that will log what zec-qt-wallet sees, so we can get to the bottom of this. Give me a few minutes.

**Comment by adityapk00** [comment]
OK, So I think I know what's happening here, and maybe in #49. zec-qt-wallet sees your zcash.conf, assumes that you've already downloaded the params and tries to start zcashd. But one of the params is missing, so zcashd doesn't start, and zec-qt-wallet is confused.

Maybe zec-qt-wallet should check the params exist even if zcash.conf is present, to ensure everything is fine.

**Comment by adityapk00** [comment]
@sonyamann Thanks for helping us get to the bottom of this issue. The problem should (hopefully) be fixed in the newest release here: https://github.com/ZcashFoundation/zec-qt-wallet/releases/tag/0.4.1

Please let us know if it works!

**Comment by rex4539** [comment]
0.4.1 crashes on launch due to hardcoded path, I believe.

```
Crashed Thread:        0
Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: @rpath/QtWidgets.framework/Versions/5/QtWidgets
  Referenced from: /Applications/zec-qt-wallet.app/Contents/MacOS/zec-qt-wallet
  Reason: image not found

```
Also, the DMG is not signed by an identified developer. This is the first time I get this error message for your builds.

**Comment by rex4539** [comment]
The missing of the `sprout-proving.key` could be related to https://github.com/zcash/zcash/issues/3652

**Comment by adityapk00** [comment]
@rex4539 Thanks for catching that. I was using a different dmg generation tool, and it looks like it doesn't work everywhere. I've reverted the 0.4.1 binary on the releases page back to the old way.

**Comment by sonyamann** [comment]
Thank you all for looking into this. Should I uninstall and re-download?

**Comment by adityapk00** [comment]
Yes please. Download the new build, install it and it should hopefully download the missing files

https://github.com/ZcashFoundation/zec-qt-wallet/releases/tag/0.4.1

**Comment by adityapk00** [comment]
Closing this since it seems like it was fixed in v0.4.1
